explainers/signed-exchange-subresource-subtitution-explainer.md

```diff
@@ -0,0 +1,455 @@
+the application/signed-exchange resource, excluding the Signature header
+field"*. So this value doesn’t change even if the publisher signs the content
+again or changes the signing key. This header-integrity value also guarantees
+the content body hasn't changed, because the signed headers are required to
+include a `Digest` header with a hash of the content body.
+
+The UA needs to check that this value of the prefetched subresource signed
+exchange is same as the header-integrity attribute of allowed-alt-sxg link
+header in the response from the publisher. This is intended to prevent the
+distributor from encoding a tracking ID into the subresource signed exchange.
+
+We can’t use the
+[SRI’s integrity](https://developer.mozilla.org/en-US/docs/Web/Security/Subresource_Integrity)
+for this purpose, because SRI’s integrity only covers the content body and not
+any of the headers. So if the UA use the SRI’s integrity value in
+allowed-alt-sxg link header, we can use the subresource signed exchanges to
+track the users by changing content-type and detecting the image loading
+failure.
+
+## Multiple subresource signed exchanges
+If there are multiple matching subresource signed exchanges (example:
+script.js.sxg and image.jpg.sxg), the UA must check that there is no error in
+the all signed exchanges (eg: sig matching, URL matching, Merkle Integrity
+error) before processing the content of the signed exchanges. This means that
+the UA can use the subresource signed exchanges only when they are defined in
+the header. This is intended to prevent the distributor from encoding a tracking
+ID into the set of subresources it prefetches. And to prevent the distributor
+from selecting a version of the subresource that isn't compatible with the
+selected version of the main resource, which might introduce a security hole.
+
+## Can’t we have a global SignedExchangeCache?
+If we can have a global SignedExchangeCache, we can use the all signed exchanges
+which have previously prefetched, even if they are not prefetched by the
+referrer page. This can improve the performance. But this introduces privacy
+issues such as
+[Timing Leaks](https://hillbrad.github.io/sri-addressable-caching/sri-addressable-caching.html#cache_timing)
+and [Deterministic History Leaks](https://hillbrad.github.io/sri-addressable-caching/sri-addressable-caching.html#history_leaks)
+described in [the document about Subresource Integrity Addressable Caching](https://hillbrad.github.io/sri-addressable-caching/sri-addressable-caching.html).
+
+To avoid these privacy issues, we introduced the limitation that we can only use
+the signed exchanges which were prefetched by the referrer page.
+
+## Can’t we merge allowed-alt-sxg to preload header?
+If we can declare the header-integrity value in the existing preload link HTTP
+header, we don’t need to introduce a new "allowed-alt-sxg" link HTTP header. But
+it becomes complicated when supporting the
```

Comment:
This might be easier to read as something like, "However, the imagesrcset attribute allows a single preload link to declare multiple different target URLs, and it's difficult to embed a `header-integrity` value for each of those URLs into the existing syntax. Instead, we use a separate link that gives a hash of the expected content for each of the possible URLs, while the preload tag continues to select which of the URLs is actually used."

(I do want to double-check that you can defend that claim that it's difficult to embed. Why couldn't it just come after the XXXw field?)